Nightly CI passes failing with "could not infer target OS" errors from ReportBinarySizes.

***Nightly builds are failing with:***
```
2022-08-24T18:23:10.9888060Z [command]/usr/local/bin/pwsh -NoLogo -NoProfile -NonInteractive -Command . '/Users/runner/work/_temp/988eecad-ed75-4795-ae69-0713e9872959.ps1'
2022-08-24T18:23:11.7682110Z ##[error]Could not infer target OS
2022-08-24T18:23:11.7699480Z ##[error]Could not infer toolchain
2022-08-24T18:23:11.7710610Z ##[error]Could not infer target platform
```

**Additional context**
THe root cause is that our OS image name changed from macOS-10.5 to macos-11.

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [ ] Bug Description Added
- [ ] Repro Steps Added
- [ ] Setup information Added

--- a/eng/pipelines/templates/stages/platform-matrix-cmakegenerate.json
+++ b/eng/pipelines/templates/stages/platform-matrix-cmakegenerate.json
@@ -1,23 +1,23 @@
 {
-    "matrix": {
-        "CmakeEnvArg": "",
-        "OSConfig": {
-            "Linux": {
-                "Pool": "azsdk-pool-mms-ubuntu-1804-general",
-                "OSVmImage": "MMSUbuntu18.04",
-                "AptDependencies": "libcurl4-openssl-dev",
-                "VCPKG_DEFAULT_TRIPLET": "x64-linux"
-            },
-            "Windows": {
-                "Pool": "azsdk-pool-mms-win-2019-general",
-                "OSVmImage": "MMS2019",
-                "VCPKG_DEFAULT_TRIPLET": "x64-windows"
-            },
-            "Mac": {
-                "Pool": "Azure Pipelines",
-                "OSVmImage": "macos-11",
-                "VCPKG_DEFAULT_TRIPLET": "x64-osx"
-            }
-        }
+  "matrix": {
+    "CmakeEnvArg": "",
+    "OSConfig": {
+      "Linux": {
+        "Pool": "azsdk-pool-mms-ubuntu-1804-general",
+        "OSVmImage": "MMSUbuntu18.04",
+        "AptDependencies": "libcurl4-openssl-dev",
+        "VCPKG_DEFAULT_TRIPLET": "x64-linux"
+      },
+      "Windows": {
+        "Pool": "azsdk-pool-mms-win-2019-general",
+        "OSVmImage": "MMS2019",
+        "VCPKG_DEFAULT_TRIPLET": "x64-windows"
+      },
+      "Mac": {
+        "Pool": "Azure Pipelines",
+        "OSVmImage": "macos-11",
+        "VCPKG_DEFAULT_TRIPLET": "x64-osx"
+      }
     }
+  }
 }

--- a/eng/pipelines/templates/stages/platform-matrix-cmakesourcegenerate.json
+++ b/eng/pipelines/templates/stages/platform-matrix-cmakesourcegenerate.json
@@ -1,23 +1,23 @@
 {
-    "matrix": {
-        "OSConfig": {
-            "Windows": {
-                "Pool": "azsdk-pool-mms-win-2019-general",
-                "OSVmImage": "MMS2019",
-                "CmakeEnvArg": ""
-            },
-            "Linux": {
-                "Pool": "azsdk-pool-mms-ubuntu-1804-general",
-                "OSVmImage": "MMSUbuntu18.04",
-                "CmakeEnvArg": "",
-                "AptDependencies": "libcurl4-openssl-dev"
-            },
-            "Mac": {
-                "Pool": "Azure Pipelines",
-                "OSVmImage": "macos-11",
-                "CmakeEnvArg": "OPENSSL_ROOT_DIR=/usr/local/opt/openssl@3 OPENSSL_INCLUDE_DIR=/usr/local/opt/openssl@3/include ",
-                "BrewDependencies": "openssl"
-            }
-        }
+  "matrix": {
+    "OSConfig": {
+      "Windows": {
+        "Pool": "azsdk-pool-mms-win-2019-general",
+        "OSVmImage": "MMS2019",
+        "CmakeEnvArg": ""
+      },
+      "Linux": {
+        "Pool": "azsdk-pool-mms-ubuntu-1804-general",
+        "OSVmImage": "MMSUbuntu18.04",
+        "CmakeEnvArg": "",
+        "AptDependencies": "libcurl4-openssl-dev"
+      },
+      "Mac": {
+        "Pool": "Azure Pipelines",
+        "OSVmImage": "macos-11",
+        "CmakeEnvArg": "OPENSSL_ROOT_DIR=/usr/local/opt/openssl@3 OPENSSL_INCLUDE_DIR=/usr/local/opt/openssl@3/include ",
+        "BrewDependencies": "openssl"
+      }
     }
+  }
 }

--- a/eng/pipelines/templates/stages/platform-matrix-live.json
+++ b/eng/pipelines/templates/stages/platform-matrix-live.json
@@ -1,152 +1,152 @@
 {
-    "displayNames": {
-        "_": ""
-    },
-    "include": [
-        {
-            "StaticConfigs": {
-                "MacOS_x64_with_unit_test": {
-                    "Pool": "Azure Pipelines",
-                    "OSVmImage": "macos-11",
-                    "VCPKG_DEFAULT_TRIPLET": "x64-osx",
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DBUILD_SAMPLES=ON",
-                    "AZURE_CORE_ENABLE_JSON_TESTS": 1,
-                    "BuildArgs": "-j 4"
-                }
-            }
-        },
-        {
-            "StaticConfigs": {
-                "_": {
-                    "VCPKG_DEFAULT_TRIPLET": "x64-linux",
-                    "BuildArgs": "-j 4",
-                    "Pool": "azsdk-pool-mms-ubuntu-1804-general",
-                    "OSVmImage": "MMSUbuntu18.04"
-                }
-            },
-            "BuildConfig": {
-                "Linux_x64_gcc5_with_unit_test": {
-                    "CmakeEnvArg": "CC=/usr/bin/gcc-5 CXX=/usr/bin/g++-5 cmake",
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON"
-                },
-                "Linux_x64_with_unit_test": {
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=Debug -DBUILD_CODE_COVERAGE=ON",
-                    "AptDependencies": "gcovr lcov",
-                    "CODE_COVERAGE": "enabled",
-                    "CODE_COVERAGE_COLLECT_ONLY": 1,
-                    "AZURE_CORE_ENABLE_JSON_TESTS": 1
-                }
-            }
-        },
-        {
-            "StaticConfigs": {
-                "_": {
-                    "BuildArgs": "-j 4",
-                    "Pool": "azsdk-pool-mms-ubuntu-2004-general",
-                    "OSVmImage": "MMSUbuntu20.04",
-                    "VCPKG_DEFAULT_TRIPLET": "x64-linux"
-                }
-            },
-            "BuildConfiguration": {
-                "Ubuntu20_x64_with_unit_test": {
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_SAMPLES=ON",
-                    "AZURE_CORE_ENABLE_JSON_TESTS": 1
-                },
-                "Ubuntu20_samples": {
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_SAMPLES=ON -DCMAKE_BUILD_TYPE=Release ",
-                    "RunSamples": 1
-                },
-                "Ubuntu20_x64_no_rtti": {
-                    "CmakeArgs": " -DBUILD_RTTI=OFF -DCMAKE_BUILD_TYPE=Release "
-                }
-            }
-        },
-        {
-            "StaticConfigs": {
-                "_": {
-                    "Pool": "azsdk-pool-mms-win-2019-general",
-                    "OSVmImage": "MMS2019",
-                    "CMAKE_GENERATOR": "Visual Studio 16 2019"
-                }
-            },
-            "BuildConfig": {
-                "Win_x86_with_unit_test_winHttp": {
-                    "VcpkgInstall": "openssl",
-                    "CMAKE_GENERATOR_PLATFORM": "Win32",
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DMSVC_USE_STATIC_CRT=ON ",
-                    "VCPKG_DEFAULT_TRIPLET": "x86-windows-static",
-                    "WindowsCtestConfig": "-C Release",
-                    "BuildArgs": "-v --parallel 8 --config Release"
-                },
-                "Win_x86_no_rtti_with_unit_test": {
-                    "VcpkgInstall": "libxml2 openssl",
-                    "CMAKE_GENERATOR_PLATFORM": "Win32",
-                    "CmakeArgs": " -DBUILD_RTTI=OFF -DBUILD_TESTING=ON -DBUILD_SAMPLES=ON -DMSVC_USE_STATIC_CRT=ON",
-                    "VCPKG_DEFAULT_TRIPLET": "x86-windows-static",
-                    "WindowsCtestConfig": "-C Release",
-                    "BuildArgs": "-v --parallel 8 --config Release"
-                },
-                "Win_x86_with_unit_test_libcurl": {
-                    "CMAKE_GENERATOR_PLATFORM": "Win32",
-                    "VCPKG_DEFAULT_TRIPLET": "x86-windows-static",
-                    "CmakeArgs": " -DBUILD_TRANSPORT_CURL=ON -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DMSVC_USE_STATIC_CRT=ON ",
-                    "BuildArgs": "-v --parallel 8"
-                },
-                "Win_x64_with_json_unit_test_winHttp": {
-                    "VcpkgInstall": "openssl",
-                    "CMAKE_GENERATOR_PLATFORM": "x64",
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DDISABLE_AZURE_CORE_OPENTELEMETRY=ON ",
-                    "BuildArgs": "-v --parallel 8 --config Release",
-                    "AZURE_CORE_ENABLE_JSON_TESTS": 1,
-                    "VCPKG_DEFAULT_TRIPLET": "x64-windows-static",
-                    "WindowsCtestConfig": "-C Release"
-                },
-                "Win_x64_with_json_unit_samples_winHttp": {
-                    "VcpkgInstall": "openssl",
-                    "VCPKG_DEFAULT_TRIPLET": "x64-windows-static",
-                    "CMAKE_GENERATOR_PLATFORM": "x64",
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DBUILD_SAMPLES=ON -DDISABLE_AZURE_CORE_OPENTELEMETRY=ON ",
-                    "BuildArgs": "-v --parallel 8 --config Release",
-                    "AZURE_CORE_ENABLE_JSON_TESTS": 1,
-                    "RunSamples": 1,
-                    "WindowsCtestConfig": "-C Release"
-                },
-                "Win_x64_with_unit_test_winHttp": {
-                    "VcpkgInstall": "openssl",
-                    "CMAKE_GENERATOR_PLATFORM": "x64",
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DMSVC_USE_STATIC_CRT=ON ",
-                    "BuildArgs": "-v --parallel 8 --config Release",
-                    "VCPKG_DEFAULT_TRIPLET": "x64-windows-static",
-                    "WindowsCtestConfig": "-C Release"
-                },
-                "Win_x64_with_unit_samples_winHttp": {
-                    "VcpkgInstall": "openssl",
-                    "VCPKG_DEFAULT_TRIPLET": "x64-windows-static",
-                    "CMAKE_GENERATOR_PLATFORM": "x64",
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DBUILD_SAMPLES=ON -DMSVC_USE_STATIC_CRT=ON ",
-                    "BuildArgs": "-v --parallel 8 --config Release",
-                    "RunSamples": 1,
-                    "WindowsCtestConfig": "-C Release"
-                },
-                "HSM_Win_x64_with_unit_test_libcurl": {
-                    "VCPKG_DEFAULT_TRIPLET": "x64-windows-static",
-                    "CMAKE_GENERATOR_PLATFORM": "x64",
-                    "CmakeArgs": " -DBUILD_TRANSPORT_CURL=ON -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DMSVC_USE_STATIC_CRT=ON ",
-                    "BuildArgs": "-v --parallel 8 --config Release",
-                    "WindowsCtestConfig": "-C Release",
-                    "KVLocation": "eastus2",
-                    "EnableHSM": 1
-                },
-                "Win_x64_with_unit_samples_libcurl": {
-                    "VcpkgInstall": "curl[winssl] openssl",
-                    "VCPKG_DEFAULT_TRIPLET": "x64-windows-static",
-                    "CMAKE_GENERATOR_PLATFORM": "x64",
-                    "CmakeArgs": " -DBUILD_TRANSPORT_CURL=ON -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON  -DBUILD_SAMPLES=ON -DMSVC_USE_STATIC_CRT=ON ",
-                    "BuildArgs": "-v --parallel 8 --config Release",
-                    "RunSamples": 1,
-                    "WindowsCtestConfig": "-C Release"
-                }
-            }
+  "displayNames": {
+    "_": ""
+  },
+  "include": [
+    {
+      "StaticConfigs": {
+        "MacOS_x64_with_unit_test": {
+          "Pool": "Azure Pipelines",
+          "OSVmImage": "macos-11",
+          "VCPKG_DEFAULT_TRIPLET": "x64-osx",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DBUILD_SAMPLES=ON",
+          "AZURE_CORE_ENABLE_JSON_TESTS": 1,
+          "BuildArgs": "-j 4"
         }
-    ]
+      }
+    },
+    {
+      "StaticConfigs": {
+        "_": {
+          "VCPKG_DEFAULT_TRIPLET": "x64-linux",
+          "BuildArgs": "-j 4",
+          "Pool": "azsdk-pool-mms-ubuntu-1804-general",
+          "OSVmImage": "MMSUbuntu18.04"
+        }
+      },
+      "BuildConfig": {
+        "Linux_x64_gcc5_with_unit_test": {
+          "CmakeEnvArg": "CC=/usr/bin/gcc-5 CXX=/usr/bin/g++-5 cmake",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON"
+        },
+        "Linux_x64_with_unit_test": {
+          "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=Debug -DBUILD_CODE_COVERAGE=ON",
+          "AptDependencies": "gcovr lcov",
+          "CODE_COVERAGE": "enabled",
+          "CODE_COVERAGE_COLLECT_ONLY": 1,
+          "AZURE_CORE_ENABLE_JSON_TESTS": 1
+        }
+      }
+    },
+    {
+      "StaticConfigs": {
+        "_": {
+          "BuildArgs": "-j 4",
+          "Pool": "azsdk-pool-mms-ubuntu-2004-general",
+          "OSVmImage": "MMSUbuntu20.04",
+          "VCPKG_DEFAULT_TRIPLET": "x64-linux"
+        }
+      },
+      "BuildConfiguration": {
+        "Ubuntu20_x64_with_unit_test": {
+          "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_SAMPLES=ON",
+          "AZURE_CORE_ENABLE_JSON_TESTS": 1
+        },
+        "Ubuntu20_samples": {
+          "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_SAMPLES=ON -DCMAKE_BUILD_TYPE=Release ",
+          "RunSamples": 1
+        },
+        "Ubuntu20_x64_no_rtti": {
+          "CmakeArgs": " -DBUILD_RTTI=OFF -DCMAKE_BUILD_TYPE=Release "
+        }
+      }
+    },
+    {
+      "StaticConfigs": {
+        "_": {
+          "Pool": "azsdk-pool-mms-win-2019-general",
+          "OSVmImage": "MMS2019",
+          "CMAKE_GENERATOR": "Visual Studio 16 2019"
+        }
+      },
+      "BuildConfig": {
+        "Win_x86_with_unit_test_winHttp": {
+          "VcpkgInstall": "openssl",
+          "CMAKE_GENERATOR_PLATFORM": "Win32",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DMSVC_USE_STATIC_CRT=ON ",
+          "VCPKG_DEFAULT_TRIPLET": "x86-windows-static",
+          "WindowsCtestConfig": "-C Release",
+          "BuildArgs": "-v --parallel 8 --config Release"
+        },
+        "Win_x86_no_rtti_with_unit_test": {
+          "VcpkgInstall": "libxml2 openssl",
+          "CMAKE_GENERATOR_PLATFORM": "Win32",
+          "CmakeArgs": " -DBUILD_RTTI=OFF -DBUILD_TESTING=ON -DBUILD_SAMPLES=ON -DMSVC_USE_STATIC_CRT=ON",
+          "VCPKG_DEFAULT_TRIPLET": "x86-windows-static",
+          "WindowsCtestConfig": "-C Release",
+          "BuildArgs": "-v --parallel 8 --config Release"
+        },
+        "Win_x86_with_unit_test_libcurl": {
+          "CMAKE_GENERATOR_PLATFORM": "Win32",
+          "VCPKG_DEFAULT_TRIPLET": "x86-windows-static",
+          "CmakeArgs": " -DBUILD_TRANSPORT_CURL=ON -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DMSVC_USE_STATIC_CRT=ON ",
+          "BuildArgs": "-v --parallel 8"
+        },
+        "Win_x64_with_json_unit_test_winHttp": {
+          "VcpkgInstall": "openssl",
+          "CMAKE_GENERATOR_PLATFORM": "x64",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DDISABLE_AZURE_CORE_OPENTELEMETRY=ON ",
+          "BuildArgs": "-v --parallel 8 --config Release",
+          "AZURE_CORE_ENABLE_JSON_TESTS": 1,
+          "VCPKG_DEFAULT_TRIPLET": "x64-windows-static",
+          "WindowsCtestConfig": "-C Release"
+        },
+        "Win_x64_with_json_unit_samples_winHttp": {
+          "VcpkgInstall": "openssl",
+          "VCPKG_DEFAULT_TRIPLET": "x64-windows-static",
+          "CMAKE_GENERATOR_PLATFORM": "x64",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DBUILD_SAMPLES=ON -DDISABLE_AZURE_CORE_OPENTELEMETRY=ON ",
+          "BuildArgs": "-v --parallel 8 --config Release",
+          "AZURE_CORE_ENABLE_JSON_TESTS": 1,
+          "RunSamples": 1,
+          "WindowsCtestConfig": "-C Release"
+        },
+        "Win_x64_with_unit_test_winHttp": {
+          "VcpkgInstall": "openssl",
+          "CMAKE_GENERATOR_PLATFORM": "x64",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DMSVC_USE_STATIC_CRT=ON ",
+          "BuildArgs": "-v --parallel 8 --config Release",
+          "VCPKG_DEFAULT_TRIPLET": "x64-windows-static",
+          "WindowsCtestConfig": "-C Release"
+        },
+        "Win_x64_with_unit_samples_winHttp": {
+          "VcpkgInstall": "openssl",
+          "VCPKG_DEFAULT_TRIPLET": "x64-windows-static",
+          "CMAKE_GENERATOR_PLATFORM": "x64",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DBUILD_SAMPLES=ON -DMSVC_USE_STATIC_CRT=ON ",
+          "BuildArgs": "-v --parallel 8 --config Release",
+          "RunSamples": 1,
+          "WindowsCtestConfig": "-C Release"
+        },
+        "HSM_Win_x64_with_unit_test_libcurl": {
+          "VCPKG_DEFAULT_TRIPLET": "x64-windows-static",
+          "CMAKE_GENERATOR_PLATFORM": "x64",
+          "CmakeArgs": " -DBUILD_TRANSPORT_CURL=ON -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DMSVC_USE_STATIC_CRT=ON ",
+          "BuildArgs": "-v --parallel 8 --config Release",
+          "WindowsCtestConfig": "-C Release",
+          "KVLocation": "eastus2",
+          "EnableHSM": 1
+        },
+        "Win_x64_with_unit_samples_libcurl": {
+          "VcpkgInstall": "curl[winssl] openssl",
+          "VCPKG_DEFAULT_TRIPLET": "x64-windows-static",
+          "CMAKE_GENERATOR_PLATFORM": "x64",
+          "CmakeArgs": " -DBUILD_TRANSPORT_CURL=ON -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON  -DBUILD_SAMPLES=ON -DMSVC_USE_STATIC_CRT=ON ",
+          "BuildArgs": "-v --parallel 8 --config Release",
+          "RunSamples": 1,
+          "WindowsCtestConfig": "-C Release"
+        }
+      }
+    }
+  ]
 }

--- a/eng/pipelines/templates/stages/platform-matrix.json
+++ b/eng/pipelines/templates/stages/platform-matrix.json
@@ -1,162 +1,162 @@
 {
-    "displayNames": {
-        "_": ""
-    },
-    "include": [
-        {
-            "OSConfiguration": {
-                "macOS-11": {
-                    "OSVmImage": "macos-11",
-                    "XCODE_VERSION": "12.5.1"
-                }
-            },
-            "StaticConfigs": {
-                "_": {
-                    "Pool": "Azure Pipelines",
-                    "BuildArgs": "-j 10",
-                    "VCPKG_DEFAULT_TRIPLET": "x64-osx",
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_TRANSPORT_CURL=ON",
-                    "PublishMapFiles": "true"
-                }
-            },
-            "BuildConfig": {
-                "debug": {
-                    "CMAKE_BUILD_TYPE": "Debug"
-                },
-                "release": {
-                    "CMAKE_BUILD_TYPE": "Release"
-                }
-            }
-        },
-        {
-            "StaticConfigs": {
-                "Windows2019": {
-                    "OSVmImage": "MMS2019",
-                    "Pool": "azsdk-pool-mms-win-2019-general",
-                    "CMAKE_GENERATOR": "Visual Studio 16 2019"
-                }
-            },
-            "TargetPlatform": {
-                "Win32Api_curl": {
-                    "CmakeArgs": " -DBUILD_TRANSPORT_CURL=ON"
-                },
-                "Win32Api_release_curl": {
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_TRANSPORT_CURL=ON -DMSVC_USE_STATIC_CRT=ON",
-                    "BuildArgs": "--parallel 8 --config Release",
-                    "PublishMapFiles": "true"
-                  },
-                "Win32Api_debug_tests": {
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_TRANSPORT_CURL=ON -DBUILD_TRANSPORT_WINHTTP=ON -DMSVC_USE_STATIC_CRT=ON",
-                    "BuildArgs": "--parallel 8 --config Debug",
-                    "PublishMapFiles": "true"
-                  }
-            },
-            "TargetArchitecture": {
-                "x86": {
-                    "CMAKE_GENERATOR_PLATFORM": "Win32",
-                    "VCPKG_DEFAULT_TRIPLET": "x86-windows-static"
-                },
-                "x64": {
-                    "CMAKE_GENERATOR_PLATFORM": "x64",
-                    "VCPKG_DEFAULT_TRIPLET": "x64-windows-static"
-                }
-            }
-        },
-        {
-            "StaticConfigs": {
-                "Windows2019": {
-                    "VcpkgInstall": "openssl",
-                    "OSVmImage": "MMS2019",
-                    "Pool": "azsdk-pool-mms-win-2019-general",
-                    "CMAKE_GENERATOR": "Visual Studio 16 2019",
-                    "PublishMapFiles": "true"
-                }
-            },
-            "TargetPlatform": {
-                "UWP_debug": {
-                  "CMAKE_SYSTEM_NAME": "WindowsStore",
-                    "CMAKE_SYSTEM_VERSION": "10.0",
-                    "CmakeArgs": " -DBUILD_TRANSPORT_WINHTTP=ON -DDISABLE_AZURE_CORE_OPENTELEMETRY=ON ",
-                    "BuildArgs": "--parallel 8 --config Debug"
-                },
-                "UWP_release": {
-                    "CMAKE_SYSTEM_NAME": "WindowsStore",
-                    "CMAKE_SYSTEM_VERSION": "10.0",
-                    "CmakeArgs": " -DBUILD_TRANSPORT_WINHTTP=ON -DDISABLE_AZURE_CORE_OPENTELEMETRY=ON ",
-                    "BuildArgs": "--parallel 8 --config Release"
-                }
-            },
-            "TargetArchitecture": {
-                "x64": {
-                    "CMAKE_GENERATOR_PLATFORM": "x64",
-                    "VCPKG_DEFAULT_TRIPLET": "x64-uwp"
-                }
-            }
-        },
-        {
-            "StaticConfigs": {
-              "Ubuntu18": {
-                  "OSVmImage": "MMSUbuntu18.04",
-                  "Pool": "azsdk-pool-mms-ubuntu-1804-general",
-                  "VCPKG_DEFAULT_TRIPLET": "x64-linux",
-                  "BuildArgs": "-j 10"
-              }
-            },
-            "BuildSettings": {
-                "gpp-5": {
-                    "AptDependencies": "g++-5",
-                    "CmakeEnvArg": "CC=/usr/bin/gcc-5 CXX=/usr/bin/g++-5 "
-                },
-                "gpp-8": {
-                    "AptDependencies": "g++-8",
-                    "CC": "/usr/bin/gcc-8",
-                    "CXX": "/usr/bin/g++-8"
-                },
-                "gpp-9": {
-                    "AptDependencies": "g++-9",
-                    "CC": "/usr/bin/gcc-9",
-                    "CXX": "/usr/bin/g++-9"
-                },
-                "included_coverage": {
-                    "AptDependencies": "gcovr lcov",
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=Debug -DBUILD_CODE_COVERAGE=ON -DCMAKE_VERBOSE_MAKEFILE=ON",
-                    "CODE_COVERAGE_COLLECT_ONLY": "1",
-                    "CODE_COVERAGE": "enabled",
-                    "PublishMapFiles": "true"
-                },
-                "included": {},
-                "included_release": {
-                    "CMAKE_BUILD_TYPE": "Release",
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON",
-                    "PublishMapFiles": "true"
-                },
-                "included_samples": {
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_SAMPLES=ON ",
-                    "PublishMapFiles": "true"
-                }
-            }
-        },
-        {
-            "StaticConfigs": {
-                "Ubuntu20": {
-                    "OSVmImage": "MMSUbuntu20.04",
-                    "Pool": "azsdk-pool-mms-ubuntu-2004-general",
-                    "VCPKG_DEFAULT_TRIPLET": "x64-linux",
-                    "BuildArgs": "-j 10",
-                    "CC": "/usr/bin/clang-11",
-                    "CXX": "/usr/bin/clang++-11",
-                    "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON",
-                    "PublishMapFiles": "true"
-                }
-            },
-            "BuildSettings": {
-                "clang-11": {
-                    "CHECK_CLANG_FORMAT": "1"
-                },
-                "included_release": {
-                    "CMAKE_BUILD_TYPE": "Release"
-                }
-            }
+  "displayNames": {
+    "_": ""
+  },
+  "include": [
+    {
+      "OSConfiguration": {
+        "macOS-11": {
+          "OSVmImage": "macos-11",
+          "XCODE_VERSION": "12.5.1"
         }
-    ]
+      },
+      "StaticConfigs": {
+        "_": {
+          "Pool": "Azure Pipelines",
+          "BuildArgs": "-j 10",
+          "VCPKG_DEFAULT_TRIPLET": "x64-osx",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_TRANSPORT_CURL=ON",
+          "PublishMapFiles": "true"
+        }
+      },
+      "BuildConfig": {
+        "debug": {
+          "CMAKE_BUILD_TYPE": "Debug"
+        },
+        "release": {
+          "CMAKE_BUILD_TYPE": "Release"
+        }
+      }
+    },
+    {
+      "StaticConfigs": {
+        "Windows2019": {
+          "OSVmImage": "MMS2019",
+          "Pool": "azsdk-pool-mms-win-2019-general",
+          "CMAKE_GENERATOR": "Visual Studio 16 2019"
+        }
+      },
+      "TargetPlatform": {
+        "Win32Api_curl": {
+          "CmakeArgs": " -DBUILD_TRANSPORT_CURL=ON"
+        },
+        "Win32Api_release_curl": {
+          "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_TRANSPORT_CURL=ON -DMSVC_USE_STATIC_CRT=ON",
+          "BuildArgs": "--parallel 8 --config Release",
+          "PublishMapFiles": "true"
+        },
+        "Win32Api_debug_tests": {
+          "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_TRANSPORT_CURL=ON -DBUILD_TRANSPORT_WINHTTP=ON -DMSVC_USE_STATIC_CRT=ON",
+          "BuildArgs": "--parallel 8 --config Debug",
+          "PublishMapFiles": "true"
+        }
+      },
+      "TargetArchitecture": {
+        "x86": {
+          "CMAKE_GENERATOR_PLATFORM": "Win32",
+          "VCPKG_DEFAULT_TRIPLET": "x86-windows-static"
+        },
+        "x64": {
+          "CMAKE_GENERATOR_PLATFORM": "x64",
+          "VCPKG_DEFAULT_TRIPLET": "x64-windows-static"
+        }
+      }
+    },
+    {
+      "StaticConfigs": {
+        "Windows2019": {
+          "VcpkgInstall": "openssl",
+          "OSVmImage": "MMS2019",
+          "Pool": "azsdk-pool-mms-win-2019-general",
+          "CMAKE_GENERATOR": "Visual Studio 16 2019",
+          "PublishMapFiles": "true"
+        }
+      },
+      "TargetPlatform": {
+        "UWP_debug": {
+          "CMAKE_SYSTEM_NAME": "WindowsStore",
+          "CMAKE_SYSTEM_VERSION": "10.0",
+          "CmakeArgs": " -DBUILD_TRANSPORT_WINHTTP=ON -DDISABLE_AZURE_CORE_OPENTELEMETRY=ON ",
+          "BuildArgs": "--parallel 8 --config Debug"
+        },
+        "UWP_release": {
+          "CMAKE_SYSTEM_NAME": "WindowsStore",
+          "CMAKE_SYSTEM_VERSION": "10.0",
+          "CmakeArgs": " -DBUILD_TRANSPORT_WINHTTP=ON -DDISABLE_AZURE_CORE_OPENTELEMETRY=ON ",
+          "BuildArgs": "--parallel 8 --config Release"
+        }
+      },
+      "TargetArchitecture": {
+        "x64": {
+          "CMAKE_GENERATOR_PLATFORM": "x64",
+          "VCPKG_DEFAULT_TRIPLET": "x64-uwp"
+        }
+      }
+    },
+    {
+      "StaticConfigs": {
+        "Ubuntu18": {
+          "OSVmImage": "MMSUbuntu18.04",
+          "Pool": "azsdk-pool-mms-ubuntu-1804-general",
+          "VCPKG_DEFAULT_TRIPLET": "x64-linux",
+          "BuildArgs": "-j 10"
+        }
+      },
+      "BuildSettings": {
+        "gpp-5": {
+          "AptDependencies": "g++-5",
+          "CmakeEnvArg": "CC=/usr/bin/gcc-5 CXX=/usr/bin/g++-5 "
+        },
+        "gpp-8": {
+          "AptDependencies": "g++-8",
+          "CC": "/usr/bin/gcc-8",
+          "CXX": "/usr/bin/g++-8"
+        },
+        "gpp-9": {
+          "AptDependencies": "g++-9",
+          "CC": "/usr/bin/gcc-9",
+          "CXX": "/usr/bin/g++-9"
+        },
+        "included_coverage": {
+          "AptDependencies": "gcovr lcov",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=Debug -DBUILD_CODE_COVERAGE=ON -DCMAKE_VERBOSE_MAKEFILE=ON",
+          "CODE_COVERAGE_COLLECT_ONLY": "1",
+          "CODE_COVERAGE": "enabled",
+          "PublishMapFiles": "true"
+        },
+        "included": {},
+        "included_release": {
+          "CMAKE_BUILD_TYPE": "Release",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON",
+          "PublishMapFiles": "true"
+        },
+        "included_samples": {
+          "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_SAMPLES=ON ",
+          "PublishMapFiles": "true"
+        }
+      }
+    },
+    {
+      "StaticConfigs": {
+        "Ubuntu20": {
+          "OSVmImage": "MMSUbuntu20.04",
+          "Pool": "azsdk-pool-mms-ubuntu-2004-general",
+          "VCPKG_DEFAULT_TRIPLET": "x64-linux",
+          "BuildArgs": "-j 10",
+          "CC": "/usr/bin/clang-11",
+          "CXX": "/usr/bin/clang++-11",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON",
+          "PublishMapFiles": "true"
+        }
+      },
+      "BuildSettings": {
+        "clang-11": {
+          "CHECK_CLANG_FORMAT": "1"
+        },
+        "included_release": {
+          "CMAKE_BUILD_TYPE": "Release"
+        }
+      }
+    }
+  ]
 }

--- a/eng/scripts/Get-BinarySizes.ps1
+++ b/eng/scripts/Get-BinarySizes.ps1
@@ -41,12 +41,15 @@ function setEnvVar($key, $value) {
 }
 
 function getTargetOs {
-    if ($OsVMImage.StartsWith('macOS')) {
+    if ($OsVMImage.StartsWith('macOS', $true, (Get-Culture).InvariantCulture)) {
         return $OsVMImage
     }
 
     if ($OsVMImage -eq "MMS2019") {
         return "win-2019"
+    }
+    if ($OsVMImage -eq "windows-2022") {
+        return "win-2022"
     }
 
     if ($OsVMImage -eq "MMSUbuntu18.04") {
@@ -57,11 +60,11 @@ function getTargetOs {
         return "ubuntu-20.04"
     }
 
-    LogError "Could not infer target OS"
+    LogError "Could not infer target OS from " $OSVmImage
 }
 
 function getTargetArchitecture {
-    if ($OSVmImage.StartsWith('macOS')) {
+    if ($OSVmImage.StartsWith('macOS', $true, (Get-Culture).InvariantCulture)) {
         return "x64"
     }
 
@@ -78,12 +81,16 @@ function getTargetArchitecture {
 }
 
 function getToolChain {
-    if ($OSVmImage.StartsWith('macOS')) {
+    if ($OSVmImage.StartsWith('macOS', $true, (Get-Culture).InvariantCulture)) {
         return "AppleClang 12"
     }
 
     if ($OSVmImage -eq "MMS2019") {
         return "MSVC"
+    }
+
+	if ($OSVmImage -eq "windows-2022") {
+        return "MSVC17"
     }
 
     if ($OSVmImage.Contains("Ubuntu")) {
@@ -98,15 +105,15 @@ function getToolChain {
         }
         return "g++-7"
     }
-    LogError "Could not infer toolchain"
+    LogError "Could not infer toolchain from " $OSVmImage and $CmakeEnvArg
 }
 
 function getTargetPlatform {
-    if ($OSVmImage.StartsWith('macOS')) {
+    if ($OSVmImage.StartsWith('macOS', $true, (Get-Culture).InvariantCulture)) {
         return "macos"
     }
 
-    if ($OSVmImage -eq 'MMS2019') {
+    if ($OSVmImage -eq 'MMS2019' -or $OsVMImage -eq "windows-2022") {
         if (!$env:CMAKE_SYSTEM_NAME -and !$CmakeArgs.Contains('WindowsStore')) {
             return 'win32'
         } elseif ($env:CMAKE_SYSTEM_NAME -eq 'WindowsStore' -or $CmakeArgs.Contains('WindowsStore')) {
@@ -118,7 +125,7 @@ function getTargetPlatform {
         return 'linux'
     }
 
-    LogError "Could not infer target platform"
+    LogError "Could not infer target platform from " $OSVmImage
 }
 
 function getBuildConfiguration {

